gc/base/ObjectScanner.hpp

```diff
@@ -112,6 +112,32 @@ class GC_ObjectScanner : public MM_BaseVirtual
 		_typeId = __FUNCTION__;
 	}
 	
+	/**
+	 * Constructor. Without leaf optimization. Context generational nursery collection.
+	 *
+	 * For marking context with leaf optimization see below:
+	 *
+	 * @param[in] env The environment for the scanning thread
+	 * @param[in] scanPtr The first slot contained in the object to be scanned
+	 * @param[in] scanMap Bit map marking object reference slots, with least significant bit mapped to slot at scanPtr
+	 * @param[in] flags A bit mask comprised of InstanceFlags
+	 * @param[in] hotFieldsDescriptor Hot fields descriptor for languages that support hot field tracking (0 if no hot fields support)
+	 */
+	GC_ObjectScanner(MM_EnvironmentBase *env, fomrobject_t *scanPtr, uintptr_t scanMap, uintptr_t flags)
+		: MM_BaseVirtual()
+		, _parentObjectPtr(NULL)
+		, _scanMap(scanMap)
+#if defined(OMR_GC_LEAF_BITS)
+		, _leafMap(0)
+#endif /* defined(OMR_GC_LEAF_BITS) */
+		, _scanPtr(scanPtr)
+		, _slotObject(env->getOmrVM(), NULL)
+		, _flags(flags | headObjectScanner)
+		, _hotFieldsDescriptor(0)
```

Comment:
couldn't you remove _hotFieldsDescriptor in this iteration?

Reply:
Yes I could. But in this case I would have unused constructor parameter. So as far as I need final cleaning Aya way to remove  _parentObjectPtr I decided to do it at the same time. Please let me know if I need to remove it right now